sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationSettingJsonConverter.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Azure.Data.AppConfiguration {
+    internal class ConfigurationSettingJsonConverter : JsonConverter<ConfigurationSetting>
+    {
+        public override ConfigurationSetting Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
```

Comment:
Is it a problem that we're not honoring the last two parameters here?  Would it make sense to throw if typeToConvert is not ConfigurationSetting?  Since there are JsonSerializerOptions we could feasibly specify here, should we also throw if this value is passed in since we don't currently honor them?  Possibly with a TODO or issue filed to track post preview 5?

It also looks like this method is never called - could we add a test for it?

Reply:
Ah, is this method called by Deserialize?  If so, nevermind about the test.

Reply:
Yes, both methods are used by `JsonSerializer`, we never call them explicitly. `JsonConverter<T>` ensures the type we get:
https://github.com/dotnet/corefx/blob/7cae92b39d2fed2e4e88e900f2e7d787ed9f6cfa/src/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs#L28

We ignore the options, but they are handled by `Utf8JsonReader` which is also provided by `JsonSerializer`

Reply:
We're not passing options to anything here, though, right?  So they wouldn't be handled if passed in?  From the look of it, though, this has a precendent in the official converters, so I think we're ok. 
 https://github.com/dotnet/corefx/tree/master/src/System.Text.Json/src/System/Text/Json/Serialization/Converters

Reply:
`JsonSerializerOptions` can be used by a complex converter for the list of user-provided converters (`JsonSerializerOptions.Converters` property), but not for the converted type itself - it is checked for us: https://github.com/dotnet/corefx/blob/7cae92b39d2fed2e4e88e900f2e7d787ed9f6cfa/src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs#L122